mobileApp/jobs/buildWhiteLabelAndroidApps.groovy

```diff
@@ -166,55 +169,90 @@ secretMap.each { jobConfigs ->
+            // Archive the following:
+            //      Generated APK file
+            //      Screenshots from emulator tests
+            //      Unit tests HTML report
+            //      Linting HTML report
             archiveArtifacts {
                 allowEmpty(false)
-                pattern('artifacts/\$APP_BASE_NAME*')
+                def screenshotPath = "${jobConfig['appBaseDir']}/edx-app-android/OpenEdXMobile/screenshots/*png"
+                pattern("artifacts/\$APP_BASE_NAME*, ${screenshotPath}")
             }
 
+            publishHtml {
+                report("\$APP_BASE_DIR/edx-app-android/OpenEdXMobile/build/reports/tests/prodDebug/") {
+                    allowMissing(false)
+                    keepAll(true)
+                    reportFiles('index.html')
+                    reportName("Unit Test Results")
+                }
+                report("\$APP_BASE_DIR/edx-app-android/OpenEdXMobile/build/outputs/") {
+                    allowMissing(false)
+                    keepAll(true)
+                    reportFiles('lint-results-prodDebug.html')
+                    reportName('Linting Results')
+                }
+            }
 
+            // Publish to HockeyApp if user says so
             configure { project ->
-                project / publishers << 'org.jenkins__ci.plugins.flexible__publish.FlexiblePublisher' (plugin: 'flexible-publish@0.15.2') {
+                project / publishers << 'org.jenkins__ci.plugins.flexible__publish.FlexiblePublisher' (
+                    plugin: 'flexible-publish@0.15.2') {
                     publishers {
-                        'org.jenkins__ci.plugins.flexible__publish.ConditionalPublisher' { 
-                            condition (class: 'org.jenkins_ci.plugins.run_condition.core.BooleanCondition', plugin: 'run-condition@1.0') {
-                                token "\${SHOULD_PUBLISH}"
-                            }
+                        'org.jenkins__ci.plugins.flexible__publish.ConditionalPublisher' {
+                            condition (
+                                class: 'org.jenkins_ci.plugins.run_condition.core.BooleanCondition',
+                                plugin: 'run-condition@1.0') {
+                                    token "\${SHOULD_PUBLISH}"
+                                }
+                            // Manually configured XML for using the HockeyApp API, because it is not present in
+                            // the Jenkins Job DSL
                             publisherList {
                                 'hockeyapp.HockeyappRecorder' (schemaVersion: '2') {
                                     applications {
-                                        'hockeyapp.HockeyappApplication' (plugin: 'hockeyapp@1.2.1', schemaVersion: '1') {
+                                        'hockeyapp.HockeyappApplication' (
+                                            plugin: 'hockeyapp@1.2.1', schemaVersion: '1') {
```

Comment:
Same here. If we change to a newer version of the plugin, will that mean we have to reprovision jobs?